platform/linux/README.md

```diff
@@ -26,25 +26,22 @@ Optional: `libsqlite3-dev` (also available as vendored dependency).
 
 ### Building with Docker
 
-You can use a Docker container to build MapLibre Native. A `Dockerfile` that installes the required dependencies when the image is built is provided in this directory.
+You can use a Docker container to build MapLibre Native. Note that you cannot build using both Docker and host methods at the same time. If you want to switch, you need to clean the repository first, e.g. by using `git clean -dxfi -e .idea -e .clwb -e .vscode`
 
 ```bash
 # Build docker image from the repo root
-docker build -t maplibre-native-image -f platform/linux/Dockerfile .
+# Specifying USER_ID and GROUP_ID allows container to create files with the same owner as the host user,
+# and avoids having to pass -u $(id -u):$(id -g) to docker run.  
+docker build \
+  -t maplibre-native-image \
+  --build-arg USER_ID=$(id -u) \
+  --build-arg GROUP_ID=$(id -g) \
+  -f platform/linux/Dockerfile \
+  platform/linux
 
-# Run docker image as the current user.
-# This ensures that the files created in the container are owned by the current user.
-docker run --rm -it -v "$PWD:/code/" -u $(id -u):$(id -g) maplibre-native-image ___any_build_command___
-
-# You can also execute build commands from inside the docker container by starting it without parameters:
-docker run --rm -it -v "$PWD:/code/" -u $(id -u):$(id -g) maplibre-native-image
-```
-
-You can safely ignore this type of message and a missing username. It happens if your linux user was not the first one created on the system.
-
-```
-groups: cannot find name for group ID ....
-I have no name!@...:/root$ `
+# Run all build commands using the docker container.
+# You can also execute build commands from inside the docker container by starting it without the build command.
```

Comment:
If we have to choose one, just include the command without the `___any_build_command___`, because it's nicer to have a valid command that people can copy and paste.

Optionally mention that you can also run build commands directly.